pkg/controller/session/model_convert_test.go

```diff
@@ -0,0 +1,155 @@
+package session_test
+
+import (
+	"github.com/aslakknutsen/istio-workspace/pkg/apis/istio/v1alpha1"
+	"github.com/aslakknutsen/istio-workspace/pkg/controller/session"
+	"github.com/aslakknutsen/istio-workspace/pkg/model"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Basic model convertion", func() {
+
+	var (
+		kind, name, aFailed, aCreated = "test", "1", "failed", "created"
+		sess                          v1alpha1.Session
+	)
+	Context("ref to status", func() {
+		var (
+			ref model.Ref
+		)
+		JustBeforeEach(func() {
+			session.RefToStatus(ref, &sess)
+
+			Expect(sess.Status).ToNot(BeNil())
+		})
+		BeforeEach(func() {
+			ref = model.Ref{
+				Name: "ref-name",
+				ResourceStatuses: []model.ResourceStatus{
+					{Kind: "test", Name: "1", Action: model.ActionCreated},
+					{Kind: "test", Name: "2", Action: model.ActionModified},
+					{Kind: "test-2", Name: "2", Action: model.ActionFailed},
+				}}
+		})
+
+		It("name mapped", func() {
+			Expect(sess.Status.Refs).To(HaveLen(1))
+			Expect(sess.Status.Refs[0].Name).To(Equal(ref.Name))
+		})
+
+		It("action mapped", func() {
+			Expect(sess.Status.Refs).To(HaveLen(1))
+			Expect(*sess.Status.Refs[0].Resources[0].Kind).To(Equal(ref.ResourceStatuses[0].Kind))
+			Expect(*sess.Status.Refs[0].Resources[0].Name).To(Equal(ref.ResourceStatuses[0].Name))
+			Expect(*sess.Status.Refs[0].Resources[0].Action).To(Equal("created"))
+
+			Expect(*sess.Status.Refs[0].Resources[1].Kind).To(Equal(ref.ResourceStatuses[1].Kind))
+			Expect(*sess.Status.Refs[0].Resources[1].Name).To(Equal(ref.ResourceStatuses[1].Name))
+			Expect(*sess.Status.Refs[0].Resources[1].Action).To(Equal("modified"))
+
+			Expect(*sess.Status.Refs[0].Resources[2].Kind).To(Equal(ref.ResourceStatuses[2].Kind))
+			Expect(*sess.Status.Refs[0].Resources[2].Name).To(Equal(ref.ResourceStatuses[2].Name))
+			Expect(*sess.Status.Refs[0].Resources[2].Action).To(Equal("failed"))
+		})
+
+		Context("ref update based on name", func() {
+			BeforeEach(func() {
+				sess = v1alpha1.Session{
+					Status: v1alpha1.SessionStatus{
+						Refs: []*v1alpha1.RefStatus{
+							{
+								Name: ref.Name,
+								Resources: []*v1alpha1.RefResource{
+									{
+										Kind: &kind, Name: &name, Action: &aFailed,
+									},
+								},
+							},
+						},
+					},
+				}
+			})
```

Comment:
![image](https://user-images.githubusercontent.com/719616/55829284-99d42e00-5b0e-11e9-8926-8aa6149117eb.png)

Reply:
Makes me wonder if some builder wouldn't make it look a bit nicer overall :)

Reply:
No clue why GitHub is optimized for horizontal space. :man_shrugging: 

![Screenshot from 2019-04-10 10-41-47](https://user-images.githubusercontent.com/132158/55864791-77c5c480-5b7d-11e9-8df4-82040ea348d3.png)